### ISSUE TITLE
Since version 3.2.0, I get "Errors applying iptables rules" when log service is on

### ISSUE THREAD

**Issue by grouvig** [ukanth/afwall#1035]
After updating from version 3..1.0 to 3.2.0, I get "Errors applying iptables rules" message.
If I install version 3.1.0 over version 3.2.0 the error continue.
If I uninstall version 3.20, install version 3.1.0 AFWall work as expected.

I didn't find anything in the logcat.
I run on Android 5.1, Magisk 20.1.

I tried running with Magisk Core only, same issue.

So right now I am stuck on version 3.1.0.
If you need i can provide more informations, tell me what you need.

**Comment by h1z1** [comment]
What is the output of iptables-save ?

**Comment by grouvig** [comment]
Well iptables-save is not installed and I don't know how to install it.

**Comment by h1z1** [comment]
Sorry assumed you rooted the device.  

tldr if you have rooted it
```
   adb shell
   su -
   iptables-save
```

You can export the rules and copy it or take a look at https://github.com/ukanth/afwall/wiki/HOWTO-Report-Bug

**Comment by grouvig** [comment]
I have rooted it.
Running "id" confirms I am root.
I can run "iptables".
But "iptables-save" is not installed.
I ran "find" on / with no result. 

Maybe I can give you some data with only iptables ?

**Comment by h1z1** [comment]
Weird, should be /system/bin/iptables-save

You can either grab the output from "Show Rules",  or "iptables -L -n -v".  It will leave out some rules but they aren't of interest atm.

**Comment by grouvig** [comment]
Well, I think I found the origin of the issue.

Rules cannot be applied when I turn on log service.
When I turn log service on, I get the error message: "kernel is missing log/nflog support".
Since then every time I apply rules I have the error message: "Errors applying iptables rules".
If I disable logs, rules are applied with success.

Log service worked on version 3.1.0 and I had it on before upgrading to 3.2.0.

Do you still need the iptables output ?

**Comment by ukanth** [comment]
Yes.

**Comment by ukanth** [comment]
Please check this version.
https://www.dropbox.com/s/p9ibfqxrasgs158/AFWall%2B_3.3.0-Test.apk?dl=0

**Comment by krbvroc1** [comment]
My AFWall+ has not been working for a week (since update to 3.2.0)....Same 'error applying iptables rules' message. It wasn't until I stumbled across this thread and disabled the log service did it start working. Even though 'log service' was enabled there was no log information or helpful details.

**Comment by grouvig** [comment]
I tested version 3.3.0_Test.
Logging service works again, there is no more "Errors applying iptables rules".

But while testing I encountered two strange behaviours.

First behaviour:
AFWall says "Rules applied with success" (with log service turned either on or off) but this is not the case.
I mean:
- With Firefox Focus allowed, I opened google with Firefox Focus -> OK
- I disallowed Firefox Focus, clicked some link, the linked opened -> KO
- I tried to apply rules again, disable/enable Firewall with no result Firefox Focus can still browse the web. -> KO

I did the same test on version 3.2.0 same results.

Whereas with version 3.1.0 it works fine
- Firefox Focus allowed, google opens -> OK
- Disable Firefox Focus, click a link and get an error from Focus -> OK
- Refresh page still error -> OK
- Allow Firefox Focus, refresh page, the page opens -> OK

Second behaviour:
With version 3.3.0_Test.
After install 94 rules applied when enabling Firewall.
Added two rules, resulting in 96 rules applied.
Disabled one rule, AFWAll still says 96 rules where applied.
Maybe this is related to First behaviour.

**Comment by ukanth** [comment]
Fixed in 3.3.0. Closing.

**Comment by ajp-dev** [comment]
I'm still having this issue in 3.3.0

Disabling logging fixes the issue.

**Comment by lolorc** [comment]
same here,
when I enable LOG, it says my kernel is lacking LOG/NFLOG support.
```
# zcat /proc/config.gz |grep NETFILTER.*LOG                         
CONFIG_NETFILTER_NETLINK_LOG=y
# CONFIG_NETFILTER_XT_TARGET_LOG is not set
CONFIG_NETFILTER_XT_TARGET_NFLOG=y
CONFIG_NETFILTER_XT_MATCH_QUOTA2_LOG=y
```

**Comment by spfanning** [comment]
I'm having the same issue in 3.4.0.
